### Problem_1: Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

The Pearson correlation coefficient, denoted by r,  measures the strength and direction of the linear relationship between two variables. It ranges from -1 to 1.

- A positive correlation (0 < r < 1) indicates that as the value of one variable increases, the value of the other variable tends to increase as well. In the case of study time and exam scores, a positive correlation would suggest that students who spend more time studying tend to score higher on the exam.
- A negative correlation (-1 < r < 0) indicates that as the value of one variable increases, the value of the other variable tends to decrease. This wouldn't be the case for study time and exam scores.
- A correlation of 0 indicates no linear relationship between the two variables.    

The closer the value of r is to 1 or -1, the stronger the linear relationship.

### Problem_2: Spearman's rank correlation is a measure of the monotonic relationship between two variables.Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [1]:
import pandas as pd

# Create a sample dataset
data = {'sleep_time': [7, 8, 6, 9, 5, 8, 7.5, 6.5, 10],
        'job_satisfaction': [7, 8.5, 6, 10, 5, 8, 7, 5.5, 9]}

df = pd.DataFrame(data)

# Calculate Spearman's rank correlation coefficient
correlation = df['sleep_time'].corr(df['job_satisfaction'], method='spearman')

# Print the correlation coefficient
print(correlation)

0.9579831932773111


### Interpretation:
- A positive Spearman's rank correlation coefficient (close to 1) indicates a positive monotonic relationship, where more sleep is associated with higher job satisfaction. 
- A negative coefficient (close to -1) indicates a negative monotonic relationship. 
- In this case, the correlation is positive (0.958) suggesting a strong positive monotonic relationship between sleep time and job satisfaction. 
- However, it's important to note that Spearman's rank correlation only measures the direction and strength of the monotonic relationship, not necessarily a linear relationship.

### Problem_3: Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [2]:
import pandas as pd

# Create sample data
data = {'exercise_hours': [3, 5, 1, 4, 2, 7, 0, 6, 8],
        'BMI': [25, 22, 28, 23, 27, 20, 30, 24, 19]}

df = pd.DataFrame(data)

# Calculate Pearson correlation coefficient
pearson_correlation = df['exercise_hours'].corr(df['BMI'], method='pearson')

# Calculate Spearman's rank correlation coefficient
spearman_correlation = df['exercise_hours'].corr(df['BMI'], method='spearman')

# Print the correlation coefficients
print("Pearson correlation coefficient:", pearson_correlation)
print("Spearman's rank correlation coefficient:", spearman_correlation)


Pearson correlation coefficient: -0.9585144756340407
Spearman's rank correlation coefficient: -0.9500000000000001


### Interpretation:
- Both coefficients are very close to -1, indicating a strong negative correlation between exercise hours and BMI. This means that as the number of hours of exercise per week increases, BMI tends to decrease.
- The high value (close to absolute value of 1) suggests a strong relationship.
- The Pearson correlation coefficient is slightly higher than the Spearman's rank correlation coefficient, but the difference is minimal. This suggests that the relationship between exercise hours and BMI is close to linear.

In conclusion, both correlation coefficients show a strong negative and likely close to linear relationship between exercise hours and BMI in this sample.

### Problem_4: A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [3]:
import pandas as pd

# Create sample data
data = {'TV_hours': [3, 5, 1, 4, 2, 7, 0, 6, 8],
        'physical_activity': [2, 1, 4, 3, 2.5, 1, 5, 3.5, 0]}

df = pd.DataFrame(data)

# Calculate Pearson correlation coefficient
pearson_correlation = df['TV_hours'].corr(df['physical_activity'], method='pearson')

# Print the correlation coefficient
print("Pearson correlation coefficient:", pearson_correlation)

Pearson correlation coefficient: -0.7940799724718948


- The Pearson correlation coefficient between TV hours and physical activity is -0.794, which indicates a negative relationship. In other words, as the number of hours spent watching television increases, the level of physical activity tends to decrease.  The strength of the correlation is moderate, so we can't say for certain that watching more television causes people to be less physically active. There could be other factors at play.

### Problem_5: A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

In [4]:
import pandas as pd
## Survey results:
data={
    'Age(Years)':[25,42,37,19,31,28],
    'Soft drink Preference':['Coke','Pepsi','Mountain dew','Coke','Pepsi','Coke']
}
df=pd.DataFrame(data)
df

Age(Years) Soft drink Preference
0          25                  Coke
1          42                 Pepsi
2          37          Mountain dew
3          19                  Coke
4          31                 Pepsi
5          28                  Coke

In [6]:
import pandas as pd

# Create the dataframe
data = {
    'Age(Years)': [25, 42, 37, 19, 31, 28],
    'Soft drink Preference': ['Coke', 'Pepsi', 'Mountain dew', 'Coke', 'Pepsi', 'Coke']
}

df = pd.DataFrame(data)
# We can explore the relationship between age and drink preference using crosstabulation
print("Crosstabulation:")
print(pd.crosstab(df['Age(Years)'], df['Soft drink Preference']))

Crosstabulation:
Soft drink Preference  Coke  Mountain dew  Pepsi
Age(Years)                                      
19                        1             0      0
25                        1             0      0
28                        1             0      0
31                        0             0      1
37                        0             1      0
42                        0             0      1


### Problem_6: A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [8]:
import pandas as pd

# Sample data (Example)
data = {
    'Sales Calls': [50, 72, 45, 68, 38, 90, 62, 40, 85],
    'Sales Made': [8, 12, 7, 10, 5, 15, 9, 6, 14]
}

df = pd.DataFrame(data)

# Calculate Pearson correlation coefficient
correlation = df['Sales Calls'].corr(df['Sales Made'], method='pearson')

# Print the correlation coefficient
print("Pearson correlation coefficient:", correlation)

Pearson correlation coefficient: 0.9898221351686111


### Interpretation:
- A positive correlation (closer to 1) indicates more sales calls are associated with more sales made per week.
- Our Correlation os 0.98 . It indicates more sales call are associated with more sales made per week.